src/Components/AffectedClustersTable/AffectedClustersTable.spec.ct.js

```diff
@@ -430,55 +430,32 @@ describe('non-empty successful affected clusters table', () => {
     ).forEach(([category, label]) => {
       SORTING_ORDERS.forEach((order) => {
         it(`${order} by ${label}`, () => {
-          const col = `td[data-label="${label}"]`;
-          const header = `th[data-label="${label}"]`;
-
-          cy.get(col).should(
-            'have.length',
-            Math.min(DEFAULT_ROW_COUNT, data.length)
-          );
-          if (order === 'ascending') {
-            cy.get(header).find('button').click();
-          } else {
-            cy.get(header).find('button').dblclick();
-          }
-
-          // add property name to clusters
-          let sortedClusters = _.cloneDeep(dataUnsorted);
-          // convert N/A timestamps as really old ones
-          sortedClusters.forEach((it) => {
-            if (it['last_checked_at'] === '') {
-              it['last_checked_at'] = '1970-01-01T01:00:00.001Z';
-            }
-            if (it.meta.cluster_version === '') {
-              it.meta.cluster_version = '0.0.0';
-            }
-          });
-
+          let sortingParameter = category;
+          // modify sortingParameters for certain values
           if (category === 'name') {
             // name sorting is case insensitive
-            category = (it) => it.name.toLowerCase();
+            sortingParameter = (it) => it.name.toLowerCase();
+          } else if (category === 'last_checked_at') {
+            sortingParameter = (it) =>
+              it.last_checked_at || '1970-01-01T01:00:00.001Z';
+          } else if (category == 'meta.cluster_version') {
+            sortingParameter = (it) =>
+              (it.meta.cluster_version || '0.0.0')
+                .split('.')
```

Comment:
I know, but I have not been able to find a way that it works with lodash sorting, so I found this workaround. The only way I know that we can use that package is using the previous implementation:

```
sortedClusters = _.map(
            category === 'meta.cluster_version'
              ? sortedClusters.sort(
                  (a, b) =>
                    (order === 'ascending' ? 1 : -1) *
                    compare(a.meta.cluster_version, b.meta.cluster_version)
                )
              : _.orderBy(
                  sortedClusters,
                  [category],
                  [order === 'ascending' ? 'asc' : 'desc']
                ),
            'name'
)```

but I see 2 issues with that:
- it is basically the same one as in the code, so if there is a mistake there, we won't be able to capture it
- it creates, in my opinion a more complex code, with an exception just for this case in which we use a different sorting function. I agree that we also need an exception in how the `sortingParameter` si computed, but (to me) it was more straightforward to put exceptions explicit in the caller function than doing some complex exception in the called function (in a shared module: `table.js`)

Reply:
you are right. I just copy the code from somewhere on the internet to pad numbers and didn't realize of the subtle thing that makes it work, and it is the `+` sign you reference in the comment below. 

Now you would get the right answer:

```
"1.5.17".split('.').map(n => +n + 100000).join('.').localeCompare("1.45.17".split('.').map(n => +n + 100000).join('.'))
```

Here is the idea. We cannot compare 2 version strings directly because `5` comes before `10`, but `'5'` comes **after** `'10'`. To fix it, we need to either compare numbers (but that requires 3 comparisons), or we add some padding to the numbers, because `'005'` comes before `'010'`. The idea of the function is to add some padding to the version strings, so we can add the comparison. The `100000` is just a value that hopefully is big enough for the test cases.

Reply:
@ikerreyes, wow, didn't now about the conversion feature of unary `+` 😅 then it's just another peculiar way of writing `parseInt`. thanks for explanation and yes, your algorithm should work 👍🏼

Reply:
TBH, I would prefer to have `parseInt` used instead, but it is a very nit pick

Reply:
Fair enough. At some point we might need to improve this to account for other versions beyond `X.Y.Z` (e.g. `X.Y.Z-release` or `X.Y.Z+build`). But we can do it later